src/components/markdown/renderers.tsx

```diff
@@ -71,3 +74,70 @@ export const InlineCodeRenderer: React.FC<ReactMarkdownCodeRendererProps> = (
 ) => {
   return <InlineCode>{props.value}</InlineCode>;
 };
+
+/*
+ * Image
+ *
+ */
+const StyledImage = styled.img`
+  width: 100%;
+`;
+
+const isCloudinaryUrl = (src: string): boolean => {
+  const url = new URL(src);
+  return (
+    url.hostname.includes('cloudinary.com') &&
+    url.pathname.startsWith('/satellytes')
+  );
+};
+
+const transformCloudinaryUrl = (src: string, width: number): string => {
+  if (width <= 0) {
+    return '';
+  }
+
+  const srcParts = src.split('/');
+  const imageId = srcParts[srcParts.length - 1];
+  const transformations = `w_${width}`;
+  return `https://res.cloudinary.com/satellytes/image/upload/${transformations}/satellytes-website/${imageId}`;
+};
+
+interface ReactMarkdownImageProps {
+  src: string;
+  alt: string;
+}
+
+export const CloudinaryImageRenderer: React.FC<ReactMarkdownImageProps> = (
+  props,
+) => {
+  // we only support cloudinary urls for now
+  if (!isCloudinaryUrl(props.src)) {
+    throw new Error(
+      'Please provide a valid image URL in markdown file. The image also needs to be hosted on Cloudinary. Your invalid URL: ' +
+        props.src,
+    );
+  }
+
+  const [imageWidth, setImageWidth] = useState(0);
+  const { width: currentWindowSize } = useWindowSize();
+
+  useDebounce(
+    () => {
+      const newImageWidth =
+        currentWindowSize >= BLOG_PAGE_MAX_WIDTH_PX
+          ? BLOG_PAGE_MAX_WIDTH_PX
+          : currentWindowSize;
+
+      // we only change the width if the picture gets better/larger
+      if (newImageWidth > imageWidth) {
+        setImageWidth(newImageWidth);
+      }
+    },
+    250,
+    [currentWindowSize],
+  );
+
+  const src = transformCloudinaryUrl(props.src, imageWidth);
```

Comment:
is it a good idea to pass any given size? will cloudinary handle this or create dozens of different sizes when you have a width of 100,101,102 etc?

Reply:
I debounced it: It will only create a new URL after 250ms and it will also only create a new URL if the image width gets larger (once you have the large image the browser can also use it when you resize to a smaller width)